src/main/java/ch/njol/skript/sections/SecLoop.java

```diff
@@ -132,20 +123,25 @@ public boolean init(Expression<?>[] exprs,
 	protected TriggerItem walk(Event event) {
 		Iterator<?> iter = currentIter.get(event);
 		if (iter == null) {
-			iter = expr instanceof Variable ? ((Variable<?>) expr).variablesIterator(event) : expr.iterator(event);
-			if (iter != null) {
-				if (iter.hasNext())
-					currentIter.put(event, iter);
-				else
-					iter = null;
+			iter = expr instanceof Variable<?> variable ? variable.variablesIterator(event) : expr.iterator(event);
+			if (iter != null && iter.hasNext()) {
+				currentIter.put(event, iter);
+				next.put(event, iter.next());
 			}
```

Comment:
this changes behavior so that iter is no longer null if hasNext is false.

Reply:
Right, but since this part of the code is only ran on startup, by it failing both of these checks, it would cause to fail the next check ending the walk

Reply:
ah i see the next.get will get it